pkg/v22/resource/app/desired.go

```diff
@@ -36,6 +41,54 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*g8s
 		return nil, microerror.Mask(err)
 	}
 
+	// TODO: Remove connection to tenant cluster once all tenant clusters use
+	// app CRs instead of chartconfig CRs.
+	//
+	//	https://github.com/giantswarm/giantswarm/issues/7402
+	//
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	if cc.Client.TenantCluster.G8s == nil {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "tenant clients not available")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		resourcecanceledcontext.SetCanceled(ctx)
+		return nil, nil
+	}
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, project.Name()),
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+	defer cancel()
+
+	// Get all configmaps in kube-system in the tenant cluster to ensure user
+	// configmaps have been migrated.
+	list, err := cc.Client.TenantCluster.K8s.CoreV1().ConfigMaps(metav1.NamespaceSystem).List(listOptions)
+	if tenant.IsAPINotAvailable(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "tenant cluster is not available yet")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		resourcecanceledcontext.SetCanceled(ctx)
+		return nil, nil
+	} else if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	if errors.Is(ctx.Err(), context.DeadlineExceeded) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "timeout getting chartconfig CRs")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		resourcecanceledcontext.SetCanceled(ctx)
+		return nil, nil
+	}
+
+	tenantConfigMaps := map[string]corev1.ConfigMap{}
+
+	for _, cm := range list.Items {
+		tenantConfigMaps[cm.Name] = cm
+	}
```

Comment:
What's purpose of this? `tenantConfigMaps` doesn't seem to be used

Reply:
@sslavic You can see it being used in https://github.com/giantswarm/cluster-operator/pull/740.

This breaks it down and moves it to v22. If there is no matching user configmap in the new location we don't create the app CR.